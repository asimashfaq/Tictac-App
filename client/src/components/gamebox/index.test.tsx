import React from 'react'
import { mount, shallow } from 'enzyme'
import sinon from 'sinon'
import GameBox from './gamebox'
import { Button } from 'antd'
describe('<GameBox />', () => {
  const onButtonClick = sinon.spy()
  it('renders the <GameBox /> components', () => {
    const ref = React.createRef<Button>()
    const wrapper = mount(<GameBox id="1" callback={onButtonClick} ref={ref} />)
    expect(wrapper).toMatchSnapshot()
    wrapper.unmount()
  })
  it('simulate the click event', () => {
    let ref = React.createRef<Button>()
    const wrapper = mount(<GameBox id="1" callback={onButtonClick} ref={ref} />)
    wrapper.find('button').simulate('click')
    expect(onButtonClick).toHaveProperty('callCount', 1)
    wrapper.unmount()
  })
})
