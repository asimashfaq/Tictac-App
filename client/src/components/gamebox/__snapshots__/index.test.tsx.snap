// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameBox /> renders the <GameBox /> components 1`] = `
<ForwardRef
  callback={[Function]}
  id="1"
>
  <Col
    span={1}
    xl={8}
  >
    <div
      className="ant-col ant-col-1 ant-col-xl-8"
    >
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        id="1"
        key="1"
        loading={false}
        onClick={[Function]}
      >
        <Wave>
          <button
            className="ant-btn"
            id="1"
            onClick={[Function]}
            type="button"
          >
            <span
              key=".0"
            >
              -
            </span>
          </button>
        </Wave>
      </Button>
    </div>
  </Col>
</ForwardRef>
`;
